app/models/form526_submission.rb

```diff
@@ -468,22 +466,11 @@ def last_remediation
   end
 
   def account
-    # first, check for an ICN on the UserAccount associated to the submission, return it if found
-    account = user_account
-    return account if account&.icn.present?
+    return user_account if user_account.icn.present?
 
-    # next, check past submissions for different UserAccounts that might have ICNs
-    past_submissions = get_past_submissions
-    account = find_user_account_with_icn(past_submissions, 'past submissions')
-    return account if account.present? && account.icn.present?
-
-    # next, check for any historical UserAccounts for that user which might have an ICN
-    user_verifications = get_user_verifications
-    account = find_user_account_with_icn(user_verifications, 'user verifications')
-    return account if account.present? && account.icn.present?
-
-    # failing all the above, default to an Account lookup
-    Account.lookup_by_user_uuid(user_uuid)
+    # check MPI for profile information using the saved EDIPI
+    mpi_response = MPI::Service.new.find_profile_by_edipi(edipi: auth_headers['va_eauth_dodedipnid'])
```

Comment:
After some more thought, we should always have a value for `edipi`, but that is no guarantee that MPI knows it, or has it associated. 

If there is a way to have it search via multiple params, or make a backup call with other info, that may be helpful

Reply:
Yes, I'm adding a backup call to MPI to query by firstname/lastname/DOB/ssn which all appear to be available in the saved headers. Even if an individual attribute like ssn is missing the attributes call can still succeed.